test/unit/remote/change.js

```diff
@@ -18,4 +18,30 @@ describe('remote change sort', () => {
     remoteChange.sort(a)
     a.should.deepEqual([parent, child])
   })
+
+  describe('sorts deleted before created for the same path', () => {
+    const deleted = {
+      doc: { path: 'parent/file' },
+      type: 'FileDeletion',
+      was: { path: 'parent/file' }
+    }
+
+    const created = {
+      doc: { path: 'parent/file' },
+      type: 'FileAddition',
+      was: null
```

Comment:
Maybe just omit `was`?

Reply:
If that's how are formed `FileAddition` changes, yes.

Reply:
You're right, it would actually be set to `null` by RemoteWatcher, my bad.